app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -88,6 +91,39 @@ class ProjectInformationActivity : AppCompatActivity() {
         }
     }
 
+    private fun setApplyButtonText(applyButton: Button, applied: Boolean?) {
+        applyButton.text = when (applied) {
+            null -> LOADING_STRING
+            true -> UNAPPLY_STRING
+            false -> APPLY_STRING
+        }
+    }
+
+    private fun setUpApplyButton(applyButton: Button) {
+        val projectId = projectVar.id
+        val userDataDatabase = Utils.getInstance(this).userDataDatabase
+        var alreadyApplied = false
+        setApplyButtonText(applyButton,null)
+        userDataDatabase.getListOfAppliedToProjects({ projectIds ->
+            alreadyApplied = projectIds.contains(projectId)
+            setApplyButtonText(applyButton, alreadyApplied)
+        },{})
+
+        applyButton.setOnClickListener {
+            userDataDatabase.applyUnapply(
+                !alreadyApplied,
+                projectVar.id,
+                {
+                    Toast.makeText(this, getString(R.string.success), Toast.LENGTH_SHORT).show()
+                    alreadyApplied = !alreadyApplied
+                    setApplyButtonText(applyButton, alreadyApplied)
+                },
+                {Toast.makeText(this, getString(R.string.failure), Toast.LENGTH_SHORT).show()}
```

Comment:
Same  as previous comment.
```suggestion
                {showToast(getString(R.string.failure))}
```